test/fetch-mocked.test.js

```diff
@@ -136,20 +155,173 @@ describe('unit tests on fetch functions', function() {
   });
 
   describe('getAllowedDatabases', function() {
-    it('should return the correct results');
-    // ...
+    var results = {};
+    beforeEach(function() {
+      // No db commands are called here.
+      results.db = makeMockDB(null, null);
+    });
+
+    it('should return all databases for which the user can list collections', function(done) {
+      results.userInfo = fixtures.USER_INFO_JOHN;
+
+      fetch.getAllowedDatabases(function(err, res) {
+        assert.equal(err, null);
+        res.sort();
+        assert.deepEqual(res, ['accounts', 'products', 'reporting', 'sales']);
+        done();
+      }, results);
+    });
+
+    it('should return empty list for users with no list collections', function(done) {
+      results.userInfo = fixtures.USER_INFO_LISTDB_ONLY;
+
+      fetch.getAllowedDatabases(function(err, res) {
+        assert.equal(err, null);
+        assert.deepEqual(res, []);
+        done();
+      }, results);
+    });
   });
+
   describe('getAllowedCollections', function() {
-    it('should return the correct results');
-    // ...
+    var results = {};
+    beforeEach(function() {
+      // No db commands are called here.
+      results.db = makeMockDB(null, null);
+    });
+
+    it('should return all collections the user info says it can access', function(done) {
+      results.userInfo = fixtures.USER_INFO_JOHN;
+
+      fetch.getAllowedCollections(function(err, res) {
+        assert.equal(err, null);
+        res.sort();
```

Comment:
the two `.sort()` are probably not needed here with 1-element arrays. what key would they sort on anyway?